city_metrix/layers/smart_cities_lulc.py

```diff
@@ -11,11 +23,194 @@ def __init__(self, land_cover_class=None, **kwargs):
+        water_osm['Value'] = 20
+        water_1m = rasterize_osm(water_osm, esa_1m)
+
+
+        # Roads
+        roads_osm = OpenStreetMap(osm_tag=roads_tag).get_data(bbox).to_crs(crs).reset_index()
+        roads_osm['lanes'] = pd.to_numeric(roads_osm['lanes'], errors='coerce')
+        # Get the average number of lanes per highway class
+        lanes = (roads_osm.drop(columns='geometry')
+                 .groupby('highway')
+                 # Calculate average and round up
+                 .agg(avg_lanes=('lanes', lambda x: np.ceil(np.nanmean(x))))
+                 )
+        # Handle NaN values in avg_lanes
+        lanes['avg_lanes'] = lanes['avg_lanes'].fillna(2)
+
+        # Fill lanes with avg lane value when missing
+        roads_osm = roads_osm.merge(lanes, on='highway', how='left')
+        roads_osm['lanes'] = roads_osm['lanes'].fillna(roads_osm['avg_lanes'])
+
+        # Add value field (30)
+        roads_osm['Value'] = 30
+
+        # Buffer roads by lanes * 10 ft (3.048 m)
+        # https://nacto.org/publication/urban-street-design-guide/street-design-elements/lane-width/#:~:text=wider%20lane%20widths.-,Lane%20widths%20of%2010%20feet%20are%20appropriate%20in%20urban%20areas,be%20used%20in%20each%20direction
+        # cap is flat to the terminus of the road
+        # join style is mitred so intersections are squared
+        roads_osm['geometry'] = roads_osm.apply(lambda row: row['geometry'].buffer(
+            row['lanes'] * 3.048,
+            cap_style=CAP_STYLE.flat,
+            join_style=JOIN_STYLE.mitre),
+            axis=1
+        )
+
+        roads_1m = rasterize_osm(roads_osm, esa_1m)
+
+
+        # TODO Building
+        # Read ULU land cover, filter to city, select lulc band
+        ulu_lulc = UrbanLandUse(band='lulc').get_data(bbox)
+        ulu_roads = UrbanLandUse(band='road').get_data(bbox)
+        # Create road mask of 50
+        # Typical threshold for creating road mask 
+        road_mask = ulu_roads >= 50
+        ulu_lulc = ulu_lulc.where(~road_mask, 6)
+        # 1-Non-residential: 0 (open space), 1 (non-res)
+        # 2-Residential: 2 (Atomistic), 3 (Informal), 4 (Formal), 5 (Housing project)
+        # 3-Roads: 6 (Roads)
+        mapping = {0: 1, 1: 1, 2: 2, 3: 2, 4: 2, 5: 2, 6: 3}
+        for from_val, to_val in mapping.items():
+            ulu_lulc = ulu_lulc.where(ulu_lulc != from_val, to_val)
+        
+        # 1-Non-residential as default
+        ulu_lulc_1m = ulu_lulc.rio.reproject(
+            dst_crs=crs,
+            shape=esa_1m.shape,
+            resampling=Resampling.nearest,
+            nodata=1
+        )
+
+        # ANBH is the average height of the built surfaces, USE THIS
+        # AGBH is the amount of built cubic meters per surface unit in the cell
+        # https://ghsl.jrc.ec.europa.eu/ghs_buH2023.php
+        anbh = (ee.ImageCollection("projects/wri-datalab/GHSL/GHS-BUILT-H-ANBH_R2023A")
```

Comment:
as mentioned in building_classifier.py, you should just make this a separate layer, and then you wouldn't need to define this twice

Reply:
Created a layer for ANBH